### ISSUE TITLE
<script type="speculationrules">: Error for missing "source" property which actually is optional

### ISSUE THREAD

**Issue by steffenweber** [validator/validator#2014]
### Summary

The checker reports an error when the "source" property is missing although it actually is optional according to the specification (both text and examples).

### URL for document that reproduces the problem

n/a

### HTML/CSS/SVG snippet that reproduces the problem

```html
<!doctype html>
<html lang="en">
<head>
<title>Title</title>
</head>
<body>
<script type="speculationrules">
{
  "prefetch": [
    {
      "urls": ["/chapters/5"]
    },
    {
      "eagerness": "moderate",
      "where": {
        "and": [
          { "href_matches": "/*" },
          { "not": { "selector_matches": ".no-prefetch" } }
        ]
      }
    }
  ]
}
</script>
</html>
```

### Checker error messages

```shell
Error: Each rule in the prefetch array within the content of a script element with a type attribute whose value is speculationrules must have a source property.
```

### Spec links/citations

https://html.spec.whatwg.org/multipage/speculative-loading.html

### Steps to reproduce

n/a

### Expected behavior

n/a

### Actual behavior

n/a

Here’s the relevant part of the spec:

[speculative-loading.html](https://github.com/user-attachments/files/24380902/speculative-loading.html)

**Comment by sideshowbarker** [comment]
Thanks much for catching this and reporting it.

It’s now fixed and deployed:

https://validator.w3.org/nu/?showsource=yes&doc=data%3Atext%2Fhtml%3Bcharset%3Dutf-8%3Bbase64%2CPCFkb2N0eXBlIGh0bWw%2BDQo8aHRtbCBsYW5nPSJlbiI%2BDQo8aGVhZD4NCjx0aXRsZT5UaXRsZTwvdGl0bGU%2BDQo8L2hlYWQ%2BDQo8Ym9keT4NCjxzY3JpcHQgdHlwZT0ic3BlY3VsYXRpb25ydWxlcyI%2BDQp7DQogICJwcmVmZXRjaCI6IFsNCiAgICB7DQogICAgICAidXJscyI6IFsiL2NoYXB0ZXJzLzUiXQ0KICAgIH0sDQogICAgew0KICAgICAgImVhZ2VybmVzcyI6ICJtb2RlcmF0ZSIsDQogICAgICAid2hlcmUiOiB7DQogICAgICAgICJhbmQiOiBbDQogICAgICAgICAgeyAiaHJlZl9tYXRjaGVzIjogIi8qIiB9LA0KICAgICAgICAgIHsgIm5vdCI6IHsgInNlbGVjdG9yX21hdGNoZXMiOiAiLm5vLXByZWZldGNoIiB9IH0NCiAgICAgICAgXQ0KICAgICAgfQ0KICAgIH0NCiAgXQ0KfQ0KPC9zY3JpcHQ%2BDQo8L2h0bWw%2B
